src/Model/Table/SentencesSentencesListsTable.php

```diff
@@ -56,6 +56,27 @@ public function getListsForSentence($sentenceId)
             ->order(['visibility', 'SentencesSentencesLists.created' => 'DESC'])
             ->all();
     }
+    public function getOwnAndPublicListsForSentence($sentenceId)
+    {
+        return $this->find()
+            ->where([
+                'sentence_id' => $sentenceId,
+                'user_id' => CurrentUser::get('id')
+            ])
+            ->orwhere([
+                'sentence_id' => $sentenceId,
+                'visibility' => 'public'
+            ])
```

Comment:
I've just noticed that [`orwhere` is deprecated since 3.5.0](https://api.cakephp.org/3.7/class-Cake.Database.Query.html#_orWhere) so this should better be written as:
````php
->where([
    'sentence_id' => $sentenceId,
    'OR' => ['user_id' => CurrentUser::get('id'), 'visibility' => 'public']
])
````